### PR TITLE
#22-As a host, I want to be able to search among energisers by name / description-JustinasBrazauskas

### LINKED ISSUE #1: Task: Create database schema for energisers
energiser: energiser_id, name, description:text

energiser_rating: rating_id, energiser_id, rating:int

### PR DESCRIPTION
**Pull request by JustinasBrazauskas**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

I created a component , SearchEnergizers. But I got stil some bugs kind like this:"the variable setSearchEnergizers is declared but not used". I'm not sure how to fix it? 
## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: client/src/pages/components/SearchEnergizers.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    const description = energizer.description.toLowerCase();
+    const searchTermLower = searchTerm.toLowerCase();
+    return name.includes(searchTermLower) || description.includes(searchTermLower);
+  });
+
+  return (
+    <div>
+      <input type="text" placeholder="Search energizers" onChange={handleSearch} />
+      {filteredEnergizers.map((energizer) => (
+        <div key={energizer.name}>
```
Comment (bertilnilsson) [inline client/src/pages/components/SearchEnergizers.js:25]: Shouldn't these be rendered with a shared component?

File: package.json [hunk 2]
```diff
@@ -77,6 +77,7 @@
 		"express": "^4.18.2",
 		"helmet": "^6.0.1",
 		"morgan": "^1.10.0",
+		"node": "^19.6.1",
```
Comment (bertilnilsson) [inline package.json:80]: `node` should not be installed as a dependency, it's the engine

File: server/api.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ router.get("/energizers", async (_, res) => {
 
 //I have only created this POST endpoint to update my database, its very incomplete as it lacks any validation, but it works and can be used as a base
 
-router.post("/energizers", (req, res) => {
+router.post("api/energizers", (req, res) => {
```
Comment (bertilnilsson) [inline server/api.js:27]: Does this not change the API url to `api/api/energizers`?

File: server/db.js [hunk 4]
```diff
@@ -12,15 +12,15 @@ const pool = new Pool({
 });
 
 export const connectDb = async () => {
-	let client;
```
Comment (bertilnilsson) [inline server/db.js:15]: This disables the database connection - included by mistake?

**Review by bertilnilsson — CHANGES_REQUESTED** [review]

**Comment by bertilnilsson** [comment]
Make sure you put the right issue number on the `Fixes #...` line in your description

File: client/src/pages/components/SearchEnergizers.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  const [energizers, setSearchEnergizers] = useState([]);
+
+  const handleSearch = (event) => {
+    setSearchTerm(event.target.value);
+  };
+
+
+
+
+  const filteredEnergizers = energizers.filter((energizer) => {
```
Comment (bertilnilsson) [inline client/src/pages/components/SearchEnergizers.js:14]: Would this not only run once when the component is first instantiated, i.e. it should it not run when you do the handleSearch (on change) above and update the state? 

Also don't you need to maintain both a state of all energizers and the filtered ones?

**Comment by JustinasBrazauskas** [comment]
closing  this because  of #46

**Pull request closed without merging**